admin/class-insert-rule-data.php

```diff
@@ -0,0 +1,119 @@
+<?php
+/**
+ * Inserts rule data about a post to the database
+ *
+ * @since 2.0.0
+ *
+ * @package Accessibility_Checker
+ */
+
+namespace EDAC\Admin;
+
+/**
+ * Class for inserting rule data into the database
+ *
+ * @since 2.0.0
+ */
+class Insert_Rule_Data {
+
+	/**
+	 * Insert rule data into database
+	 *
+	 * @since 2.0.0
+	 *
+	 * @param object $post     The post object.
+	 * @param string $rule     The rule.
+	 * @param string $ruletype The rule type.
+	 * @param string $rule_obj The object.
+	 *
+	 * @return void|int
+	 */
+	public function insert( object $post, string $rule, string $ruletype, string $rule_obj ) {
+
+		if ( ! $post || ! $rule || ! $ruletype || ! $rule_obj ) {
+			return;
+		}
+
+		global $wpdb;
+		$table_name = $wpdb->prefix . 'accessibility_checker';
+
+		// set up rule data array.
+		$rule_data = array(
+			'postid'        => $post->ID,
+			'siteid'        => get_current_blog_id(),
+			'type'          => $post->post_type,
+			'rule'          => $rule,
+			'ruletype'      => $ruletype,
+			'object'        => esc_attr( $rule_obj ),
+			'recordcheck'   => 1,
+			'user'          => get_current_user_id(),
+			'ignre'         => 0,
+			'ignre_user'    => null,
+			'ignre_date'    => null,
+			'ignre_comment' => null,
+			'ignre_global'  => 0,
+		);
+
+		// return if revision.
+		if ( 'revision' === $rule_data['type'] ) {
+			return;
+		}
+
+		// Check if exists.
+		// phpcs:ignore WordPress.DB.DirectDatabaseQuery.DirectQuery, WordPress.DB.DirectDatabaseQuery.NoCaching -- Using direct query for adding data to database, caching not required for one time operation.
+		$results = $wpdb->get_results(
+			$wpdb->prepare(
+				'SELECT postid, ignre FROM %i where type = %s and postid = %d and rule = %s and object = %s and siteid = %d',
+				$table_name,
+				$rule_data['type'],
+				$rule_data['postid'],
+				$rule_data['rule'],
+				$rule_data['object'],
+				$rule_data['siteid']
+			),
+			ARRAY_A
+		);
+
+		// Loop existing records.
+		if ( $results ) {
+			foreach ( $results as $row ) {
+
+				// if being ignored, don't overwrite value.
+				if ( true === (bool) $row['ignre'] ) {
+					$rule_data['ignre'] = 1;
+				}
+
+				// update existing record.
+				// phpcs:ignore WordPress.DB.DirectDatabaseQuery.DirectQuery, WordPress.DB.DirectDatabaseQuery.NoCaching -- Using direct query for adding data to database, caching not required for one time operation.
+				$wpdb->query(
+					$wpdb->prepare(
+						'UPDATE %i SET recordcheck = %d, ignre = %d  WHERE siteid = %d and postid = %d and rule = %s and object = %s and type = %s',
+						$table_name,
+						1,
+						$rule_data['ignre'],
+						$rule_data['siteid'],
+						$rule_data['postid'],
+						$rule_data['rule'],
+						$rule_data['object'],
+						$rule_data['type']
+					)
+				);
+
+			}
+		}
+
+		// Insert new records.
+		if ( ! $results ) {
+
+			// filter post types.
+			$rule_data = apply_filters( 'edac_filter_insert_rule_data', $rule_data );
```

Comment:
@pattonwebz, should we sanitize this data to ensure it's been filtered in a safe manner?

Example:
```php
// Apply filters to allow modification of the rule data.
$rule_data = apply_filters( 'edac_filter_insert_rule_data', $rule_data );

// Sanitize the filtered data.
$rule_data_sanitized = array(
    'postid'        => absint( $rule_data['postid'] ),
    'siteid'        => absint( $rule_data['siteid'] ),
    'type'          => sanitize_text_field( $rule_data['type'] ),
    'rule'          => sanitize_text_field( $rule_data['rule'] ),
    'ruletype'      => sanitize_text_field( $rule_data['ruletype'] ),
    'object'        => sanitize_text_field( $rule_data['object'] ),
    'recordcheck'   => absint( $rule_data['recordcheck'] ),
    'user'          => absint( $rule_data['user'] ),
    'ignore'        => absint( $rule_data['ignore'] ), // Note: Fixed typo 'ignre' to 'ignore'
    'ignore_user'   => isset( $rule_data['ignore_user'] ) ? absint( $rule_data['ignore_user'] ) : null,
    'ignore_date'   => isset( $rule_data['ignore_date'] ) ? sanitize_text_field( $rule_data['ignore_date'] ) : null,
    'ignore_comment'=> isset( $rule_data['ignore_comment'] ) ? sanitize_text_field( $rule_data['ignore_comment'] ) : null,
    'ignore_global' => absint( $rule_data['ignore_global'] ),
);

// Now, $rule_data_sanitized contains the sanitized data ready for insertion.
```

Reply:
I see that you fixed the `ignre` to `ignore,` but we can't do that without first remapping the columns in the table. Would you like me to make an issue to tackle that in the future?